### DISCUSSION TITLE
Input layout

### DISCUSSION THREAD

**Discussion by ryanmckaytx** [observablehq/framework#1631] [Q&A] [answered]
````
```js
const selected_merchant = view(
  Inputs.select(merchants, {
    label: "Store",
  })
);
const start_date = view(
  Inputs.date({label: "Start", min: "2023-01-01", max: "2023-12-31", value: "2023-10-01"})
);
const end_date = view(
  Inputs.date({label: "End", min: "2023-01-01", max: "2023-12-31", value: "2023-12-31"})
);
```
````
renders as:
<img width="377" alt="Screenshot 2024-09-02 at 4 13 31 PM" src="https://github.com/user-attachments/assets/0e32559a-a059-48e2-a809-cd80c676c9c2">

how could you get it to render more like:
<img width="753" alt="observable-inputs2" src="https://github.com/user-attachments/assets/b8d13ee8-9051-4564-99c9-71ddea35f216">

**Comment by Fil** [comment, marked as answer]
With a bit of CSS:

```
<style>
  .inline {
    display: flex;
    flex-wrap: wrap;
    gap: 20px;
    max-width: 640px;
  }
  .inline form {
    max-width: 200px;
    display: flow;
  }
</style>
```

It can also be nice to return this as a combined input, with [Inputs.form](https://github.com/observablehq/inputs?tab=readme-ov-file#inputsforminputs-options)

See https://observablehq.observablehq.cloud/pangea/support/input-layout for the implementation.

**Reply by ryanmckaytx** [reply]
worked perfectly, thank you so much!

**Reply by ryanmckaytx** [reply]
How would you package all of that as a component?

**Reply by Fil** [reply]
See https://observablehq.com/framework/lib/inputs to import `Inputs`. Then probably attach the styles inline instead of via a stylesheet? Let me know if you try it.
